Orange/widgets/visualize/owscatterplotgraph.py

```diff
@@ -403,6 +409,7 @@ def __init__(self, scatter_widget, parent=None, view_box=ViewBox):
         self._tooltip_delegate = EventDelegate(self.help_event)
         self.plot_widget.scene().installEventFilter(self._tooltip_delegate)
         self.view_box.sigTransformChanged.connect(self.update_density)
+        self.view_box.sigRangeChangedManually.connect(self.update_labels)
```

Comment:
I switched from `sigRangeChangedManually` to avoid updating labels twice on `reset_graph` or `update_coordinates`.

Reply:
I think you'll have to go back to sigRangeChanged - when resetting the zoo, the labels remain visible.